pkg/machinery/config/configloader/internal/decoder/decoder.go

```diff
@@ -124,6 +128,8 @@ func decode(manifest *yaml.Node) (target config.Document, err error) {
 	}
 
 	switch {
+	case version == "":
```

Comment:
this fixes a bug when `apiVersion` is not defined properly in the config document and the error was not clear before

Reply:
I wonder if that should be moved down, as it might break loading config patches with v1alpha1 strategic merge (???) not sure

but if the version is not defined, it should actually return an error from `registry.New()` below?

Reply:
I think this is still important

Reply:
moved to #8270